contracts/filters/paraswap-v5/ParaswapV5Filter.sol

```diff
@@ -0,0 +1,80 @@
+// Copyright (C) 2022  Argent Labs Ltd. <https://argent.xyz>
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.3;
+
+import "../BaseFilter.sol";
+
+contract ParaswapV5Filter is BaseFilter {
+
+    address private immutable agsAddress;
+    address private immutable augustusAddress;
+
+    constructor(address _agsAddress, address _augustusAddress) {
+        agsAddress = _agsAddress;
+        augustusAddress = _augustusAddress;
+    }
+
+    function isValid(address _wallet, address /*_spender*/, address _to, bytes calldata _data) external view override returns (bool valid) {
+        // disallow ETH transfer
+        if (_data.length < 4) {
+            return false;
+        }
+
+        // only allow calls to AugustusSwapper
+        if (_to != augustusAddress) {
+            return false;
+        }
+
+        // 65 is length of signature, 32 is length of deadline
+        bytes calldata swapData = _data[:_data.length - 65 - 32];
```

Comment:
Not sure if the compiler takes care of that but it might be more efficient to use `97` instead of `65 + 32`.

Also, maybe storing `data.length` in memory might be better than getting it from calldata multiple times.

Reply:
I'm 99% sure the solidity compiler does `constant folding`, it's a very basic optimisation